src/components/SVGViewer/SVGViewer.tsx

```diff
@@ -104,7 +121,22 @@ export class SVGViewer extends React.Component<SvgViewerProps, SvgViewerState> {
       this.setCustomClasses();
     }
 
-    if (prevProps.documentId !== this.props.documentId) {
+    // If supplied props has changed from `documentId` to `file` or vice versa
+    if (isDocumentIdProps(prevProps) !== isDocumentIdProps(this.props)) {
```

Comment:
Mb we can replace it with next solution:
```typescript
const { documentId: prevDocId } = prevProps as SvgViewerDocumentIdProps;
const { documentId: curDocId } = this.props as SvgViewerDocumentIdProps;
const { file: prevFile } = prevProps as SvgViewerFileProps;
const { file: currFile } = this.props as SvgViewerFileProps;

if (
 (curDocId && curDocId !== prevDocId) ||
 (currFile && currFile !== prevFile)
) {
  this.presetSVG();
}
```

Reply:
I guess the types in this case is simple enough to just cast :P

Reply:
what about the case where they gave a `documentId` before and then `file` or vice versa?

Reply:
It will still work. `documentId` become `undefined` and `file` will get some value at the same time.

Reply:
but we can also add test for this case

Reply:
Fair fair good point

Reply:
this looks much better